source/module_esolver/esolver_ks.cpp

```diff
@@ -670,6 +671,15 @@ void ESolver_KS<T, Device>::iter_finish(UnitCell& ucell, const int istep, int& i
     {
         dkin = p_chgmix->get_dkin(pelec->charge, PARAM.inp.nelec);
     }
+
+    double pseudopot_energy = 0.;                   // electron-ion interaction energy
+    for (int is = 0; is < PARAM.inp.nspin; ++is)
+    {
+        pseudopot_energy += BlasConnector::dot(this->pw_rho->nrxx, this->pelec->pot->get_fixed_v(), 1, pelec->charge->rho[is], 1)
```

Comment:
Could you separate this part as a function and move it to elecstate, for example?

Reply:
I have moved this part to `ElecState::get_local_pp_energy`.